jax/_src/lax/linalg.py

```diff
@@ -1973,25 +2056,38 @@ def qr_jvp_rule(primals, tangents, *, full_matrices):
   do = do + I * (qt_dx_rinv - qt_dx_rinv.real.astype(qt_dx_rinv.dtype))
   dq = q @ (do - qt_dx_rinv) + dx_rinv
   dr = (qt_dx_rinv - do) @ r
+  if pivoting:
+    dp = ad_util.Zero.from_primal_value(p[0])
+    return (q, r, p), (dq, dr, dp)
```

Comment:
It seems like pivoting should affect the tangents as well. Can you add more details about why the JVP rule doesn't need to change with pivoting?

Reply:
I believe the impact of pivoting on the tangents is captured entirely by the `q` and `r` values that are returned from `qr_p.bind(...)`, and that tangent validity is subject to, at worst, the same assumptions as the non-pivoted case (that `x` is  of full column-rank `k = min(m, n)`).

However, this is not my area of expertise, so I may be missing something.

Reply:
In the case where `pivoting=True`, I would expect `dx` to be indexed by `p`. If I understand the behavior properly, `x[..., p] = q @ r`, so we'll want to use `dx[..., p]` when propagating the tangents, right?

Relatedly: it would be good to add some test coverage for AD when `pivoting=True`. This might also help pointing us of the right thing to do here! Here are the current AD tests for QR which might be a useful place to start:

https://github.com/jax-ml/jax/blob/0f4677bcf68f92364d7e8bf9299ee659e2745c20/tests/linalg_test.py#L1025-L1032

Reply:
Ah yes, you are right! Because the Numpy API doesn't support pivoting, I will add a `check_jvp` statement to the end of `testScipyQrModes`.